src/App.tsx

```diff
@@ -1,15 +1,57 @@
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
+import { PersonData } from './types/PersonData';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPersonData, setSelectedPersonData] =
+    useState<PersonData | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const filteredPeople = useMemo(() => {
```

Comment:
This is a performance issue. You are filtering `peopleFromServer` on every render which is not efficient. Instead, you should filter the people list only when `appliedQuery` changes. You could move the filter logic into the `useEffect` hook that depends on `appliedQuery`.